aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -384,30 +386,85 @@ public void deletePrefix(String prefix) {
     deleteFiles(() -> Streams.stream(listPrefix(prefix)).map(FileInfo::location).iterator());
   }
 
+  /**
+   * Returns the {@link S3Client} that is configured for this {@link org.apache.iceberg.io.FileIO}
+   * instance.
+   *
+   * @deprecated since 1.10.0, will be removed in 1.11.0; use {@link
+   *     S3FileIO#clientForStoragePath(String)} instead.
+   */
+  @Deprecated
   public S3Client client() {
-    if (client == null) {
-      synchronized (this) {
-        if (client == null) {
-          client = s3.get();
-        }
+    return clientForStoragePath("s3").s3();
+  }
+
+  /**
+   * Returns the {@link S3AsyncClient} that is configured for this {@link
+   * org.apache.iceberg.io.FileIO} instance.
+   *
+   * @deprecated since 1.10.0, will be removed in 1.11.0; use {@link
+   *     S3FileIO#clientForStoragePath(String)} instead.
+   */
+  @Deprecated
+  public S3AsyncClient asyncClient() {
+    return clientForStoragePath("s3").s3Async();
+  }
+
+  public PrefixedS3Client clientForStoragePath(String storagePath) {
+    PrefixedS3Client client;
+    String matchingPrefix = "s3";
```

Comment:
Would it make sense to move all the references of "s3" to a constant in this file `ROOT_PREFIX`?

Reply:
yep good idea, I've done that